tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CustomizedCodeUpdateTool/TspClientUpdateTool.cs

```diff
@@ -0,0 +1,639 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.ComponentModel;
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Models;
+using ModelContextProtocol.Server;
+using Azure.Sdk.Tools.Cli.Services;
+using System.Text.Json;
+
+namespace Azure.Sdk.Tools.Cli.Tools;
+
+[McpServerToolType, Description("Update customized SDK code after TypeSpec regeneration: creates/continues a session, diffs old vs new generated code, maps API changes to impacted customization files, proposes patches, applies them, and reports status.")]
+public class TspClientUpdateTool : MCPTool
+{
+    private readonly ILogger<TspClientUpdateTool> logger;
+    private readonly IOutputService output;
+
+    // --- CLI options/args ---
+    private readonly Argument<string> _specPathArg = new(name: "spec-path", description: "Path to the .tsp specification file") { Arity = ArgumentArity.ExactlyOne };
+    private readonly Option<string> _sessionIdOpt = new(["--session", "-s"], () => string.Empty, "Existing session id (omit to create a new session)");
+    private readonly Option<string> _generatedOldOpt = new(["--old-gen"], () => string.Empty, "Path to previously generated code (optional)");
+    private readonly Option<string> _generatedNewOpt = new(["--new-gen"], () => string.Empty, "Target path for new generated code (optional, temp path if omitted)");
+    private readonly Option<bool> _dryRunOpt = new(["--dry-run"], () => false, "Do not write any changes when applying patches");
+    private readonly Option<string> _filesOpt = new(["--files"], () => string.Empty, "Comma-separated list of customization files to target (defaults to all impacted)");
+
+    // Non-static session store to avoid global mutable state
+    private readonly Dictionary<string, UpdateSessionState> _sessions = new(StringComparer.OrdinalIgnoreCase);
+    private static readonly string SessionDir = Path.Combine(Directory.GetCurrentDirectory(), ".tspupdate-sessions");
+
+    public TspClientUpdateTool(ILogger<TspClientUpdateTool> logger, IOutputService output)
+    {
+        this.logger = logger;
+        this.output = output;
+    }
+
+    public override Command GetCommand()
+    {
+        var root = new Command("customized-update", "Workflow for updating customized TypeSpec-generated client code");
```

Comment:
We will probably change it later, but to start I think we can put this and the other tsp tools underneath `tsp` or `tsp-client` or whatever makes sense.

https://github.com/Azure/azure-sdk-tools/blob/main/tools/azsdk-cli/docs/new-tool.md#step-2-define-command-group-and-structure

Reply:
>I think we can put this and the other tsp tools underneath tsp or tsp-client or whatever makes sense.

 I wanted to use a new grouping too but didnt find one for tsp-client related updates and just used eng as fallback. Will change this to `tsp`